internal/configs/bootnode/config.go

```diff
@@ -0,0 +1,364 @@
+// Package nodeconfig includes all the configuration variables for a node.
+// It is a global configuration for node and other services.
+// It will be included in node module, and other modules.
+package bootnode
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	bls_core "github.com/harmony-one/bls/ffi/go/bls"
+	"github.com/harmony-one/harmony/crypto/bls"
+	shardingconfig "github.com/harmony-one/harmony/internal/configs/sharding"
+	"github.com/harmony-one/harmony/internal/params"
+	"github.com/harmony-one/harmony/multibls"
+	"github.com/harmony-one/harmony/shard"
+	"github.com/harmony-one/harmony/webhooks"
+	p2p_crypto "github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/pkg/errors"
+)
+
+// Role defines a role of a node.
+type Role byte
+
+// All constants for different node roles.
+const (
+	Unknown Role = iota
+	BootNode
+)
+
+func (role Role) String() string {
+	switch role {
+	case BootNode:
+		return "BootNode"
+	default:
+		return "Unknown"
+	}
+}
+
+// NetworkType describes the type of Harmony network
+type NetworkType string
+
+// Constants for NetworkType
+// TODO: replace this with iota. Leave the string parsing in command line
+const (
+	Mainnet   = "mainnet"
+	Testnet   = "testnet"
+	Pangaea   = "pangaea"
+	Partner   = "partner"
+	Stressnet = "stressnet"
+	Devnet    = "devnet"
+	Localnet  = "localnet"
+)
+
+// ChainConfig returns the chain configuration for the network type.
+func (t NetworkType) ChainConfig() params.ChainConfig {
+	switch t {
+	case Mainnet:
+		return *params.MainnetChainConfig
+	case Pangaea:
+		return *params.PangaeaChainConfig
+	case Partner:
+		return *params.PartnerChainConfig
+	case Stressnet:
+		return *params.StressnetChainConfig
+	case Localnet:
+		return *params.LocalnetChainConfig
+	default:
+		return *params.TestnetChainConfig
+	}
+}
+
+func (n NetworkType) String() string {
+	if n == "" {
+		return Testnet // default to testnet
+	}
+	return string(n)
+}
+
+// Global is the index of the global node configuration
+const (
```

Comment:
bootnode doesn't care of the number of shards here. So let's remove that part

Reply:
removed